server/src/main/java/org/opensearch/action/admin/indices/scale/searchonly/ScaleIndexResponse.java

```diff
@@ -0,0 +1,137 @@
+     *
+     * @param out the output stream to write to
+     * @throws IOException if there is an I/O error during serialization
+     */
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeList(new ArrayList<>(nodeResponses));
+        out.writeBoolean(hasFailures);
+        out.writeOptionalString(failureReason);
+    }
+
+    /**
+     * Indicates whether any shards reported conditions that would prevent
+     * safely finalizing the scale operation.
+     *
+     * @return true if any shard has uncommitted operations or needs sync, false otherwise
+     */
+    boolean hasFailures() {
+        return hasFailures;
+    }
+
+    /**
+     * Builds a detailed description of failure reasons if the scale operation cannot proceed.
+     * <p>
+     * This method constructs a human-readable string explaining which shards on which nodes
+     * reported conditions that prevent the scale operation from being finalized, including
+     * whether they have uncommitted operations or need additional synchronization.
+     *
+     * @return a detailed failure description, or null if no failures were detected
+     */
+    String buildFailureReason() {
+        if (!hasFailures) {
+            return null;
+        }
+        StringBuilder reason = new StringBuilder();
+        for (ScaleIndexNodeResponse nodeResponse : nodeResponses) {
+            for (ScaleIndexShardResponse shardResponse : nodeResponse.getShardResponses()) {
+                if (shardResponse.hasUncommittedOperations() || shardResponse.needsSync()) {
+                    reason.append("Shard ")
+                        .append(shardResponse.getShardId())
+                        .append(" on node ")
+                        .append(nodeResponse.getNode())
+                        .append(": ");
+                    if (shardResponse.hasUncommittedOperations()) {
+                        reason.append("has uncommitted operations ");
+                    }
+                    if (shardResponse.needsSync()) {
+                        reason.append("needs sync ");
+                    }
+                    reason.append("; ");
+                }
+            }
+        }
+        return reason.toString();
+    }
+
+    /**
+     * Converts this response to XContent format for API responses.
+     * <p>
+     * The generated content includes:
+     * <ul>
+     *   <li>Whether any failures were detected</li>
+     *   <li>Detailed failure reasons if applicable</li>
+     * </ul>
+     *
+     * @param builder the XContentBuilder to use
+     * @param params  parameters for XContent generation
+     * @return the XContentBuilder with response data added
+     * @throws IOException if there is an error generating the XContent
+     */
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        builder.startObject();
+        builder.field("has_failures", hasFailures);
```

Comment:
Do we need the `has_failures` property? Or can we just rely on `failure_reason` being non-null?

Reply:
So the relation is `has_failures` is true if and only if `failure_reason` is present, coming from the method `buildFailureReason` part of the same class.

Reply:
Yes we can remove the `builder.field("has_failures", hasFailures);`.